src/super_gradients/training/transforms/pipeline_adaptors.py

```diff
@@ -83,6 +135,34 @@ def post_transforms_processing(self, sample):
             sample = SegmentationSample(image=Image.fromarray(sample["image"]), mask=Image.fromarray(sample["mask"]))
         elif self.sample_type == SampleType.DEPTH_ESTIMATION:
             sample = DepthEstimationSample(image=sample["image"], depth_map=sample["mask"])
+        elif self.sample_type == SampleType.POSE_ESTIMATION:
+
+            if len(sample["bboxes"]) == 0:
+                sample["bboxes"] = np.zeros((0, 4))
+            if len(sample["is_crowd"]) == 0:
+                sample["is_crowd"] = np.zeros((0))
+
+            bboxes_xywh = xyxy_to_xywh(bboxes=np.array(sample["bboxes"]), image_shape=sample["image"].shape)
+
+            # Update value of keypoints that are not inside the image anymore.
+            h, w = sample["image"].shape[1], sample["image"].shape[0]
+            keypoints = np.array(
+                [keypoint if (0, 0) <= (keypoint[0], keypoint[1]) < (w, h) else (keypoint[0], keypoint[1], 0) for keypoint in sample["keypoints"]]
+            )
+            keypoints = keypoints.reshape(-1, sample["n_joints"], 3)
+
+            keypoints = keypoints[sample["labels"]]  # remvoe the ones associated with a bbox that was removed.
```

Comment:
I think we also should filter `is_crowd[sample["labels"]]` if it's present.

Reply:
Actually no because `is_crowd` is `label` of `bbox_params` so it's filtered along with `labels`. I double checked and moving to `is_crowd[sample["labels"]]` would even make the code crash (filtering twice so there is index error)

Reply:
@BloodAxe About your idea of using `sample.filter_by_mask(labels)`.
It works if we won't have the `is_crowd` param. This is `is_crowd` is already filtered by albumentation, so it fails if we use `filter_by_mask` since it will try to filter it once more and fail (same error as in my previous message).
Basically, `bboxes` and `is_crowd` are mapped as `bbox`, so both are filtered together when a bbox is invalid. `keypoints` is the only remaining param that we need to filter.